dlt/normalize/items_normalizers.py

```diff
@@ -0,0 +1,131 @@
+                            table_name
+                        )
+                    # get current columns schema
+                    columns = column_schemas.get(table_name)
+                    if not columns:
+                        columns = schema.get_table_columns(table_name)
+                        column_schemas[table_name] = columns
+                    # store row
+                    # TODO: it is possible to write to single file from many processes using this: https://gitlab.com/warsaw/flufl.lock
+                    load_storage.write_data_item(
+                        load_id, schema_name, table_name, row, columns
+                    )
+                    # count total items
+                    items_count += 1
+                    increase_row_count(row_counts, table_name, 1)
+            signals.raise_if_signalled()
+        return schema_update, items_count, row_counts
+
+    def __call__(
+        self,
+        extracted_items_file: str,
+        load_storage: LoadStorage,
+        normalize_storage: NormalizeStorage,
+        schema: Schema,
+        load_id: str,
+        root_table_name: str,
+    ) -> Tuple[List[TSchemaUpdate], int, TRowCount]:
+        schema_updates: List[TSchemaUpdate] = []
+        row_counts: TRowCount = {}
+        with normalize_storage.storage.open_file(extracted_items_file) as f:
+            # enumerate jsonl file line by line
+            items_count = 0
+            for line_no, line in enumerate(f):
+                items: List[TDataItem] = json.loads(line)
+                partial_update, items_count, r_counts = self._normalize_chunk(
+                    load_storage, schema, load_id, root_table_name, items
+                )
+                schema_updates.append(partial_update)
+                merge_row_count(row_counts, r_counts)
+                logger.debug(
+                    f"Processed {line_no} items from file {extracted_items_file}, items {items_count}"
+                )
+
+        return schema_updates, items_count, row_counts
+
+
+class ParquetItemsNormalizer(ItemsNormalizer):
+    def __call__(
+        self,
+        extracted_items_file: str,
+        load_storage: LoadStorage,
+        normalize_storage: NormalizeStorage,
+        schema: Schema,
+        load_id: str,
+        root_table_name: str,
+    ) -> Tuple[List[TSchemaUpdate], int, TRowCount]:
+        from dlt.common.libs import pyarrow
+        with normalize_storage.storage.open_file(extracted_items_file, "rb") as f:
+            items_count = pyarrow.get_row_count(f)
+        target_folder = load_storage.storage.make_full_path(os.path.join(load_id, LoadStorage.NEW_JOBS_FOLDER))
+        # normalize_storage.parse_normalize_file_name
+        parts = NormalizeStorage.parse_normalize_file_name(extracted_items_file)
+        new_file_name = load_storage.build_job_file_name(parts.table_name, parts.file_id, with_extension=False) + ".parquet"
```

Comment:
heh interesting if that is going to work :) in theory we can have some files to be jsonl/INSERT and some parquet and the destination may handle that.

IMO you should create a valid `load_storage` with a proper extension and pass it here.